pkg/controller/plan/kubevirt.go

```diff
@@ -630,6 +615,17 @@ func (r *KubeVirt) createVolumesForOvirt(vm ref.Ref) (pvcNames []string, err err
 			return
 		}
 		pvcNames = append(pvcNames, pvc.Name)
+
+		err = k8sutil.SetOwnerReference(pvc, populatorCr, r.Scheme())
```

Comment:
who removes it when the migration is removed? i think nobody (unless they will archive the plan and it will clean it up). i think it's better to have an owner.

Reply:
the ownership is just another way of cleanup. to enter the CleanUp, you need to either archive or cancel the migration. many users may just delete it without these steps. in that case - if the migration passed and the plan deleted, we will have leftovers, but it still somehow relate to the PVC-VM. in case they are deleted we are really not interested/need these so the ownership will take care of it.

Reply:
> the ownership is just another way of cleanup. 

does it imply that we would still initiate calls to remove the CRs when the migration is removed?

> to enter the CleanUp, you need to either archive or cancel the migration. many users may just delete it without these steps. in that case - if the migration passed and the plan deleted, we will have leftovers, but it still somehow relate to the PVC-VM.

while we don't have empirical data on the amount of users remove plans without archiving them, this should have been addressed already by the message that appears when deleting a plan without archiving it first - there's a bold statement that this would result in leaving behind leftovers.

> in case they are deleted we are really not interested/need these so the ownership will take care of it.

yes, I see why you went this way - my issue with that is that the pvc is not really the owner of the CR.
usually when you have true owner, you could specify it when creating a resource. for instance, when creating a migration you could say that the plan is its owner - the plan already exists, and the migration is derived by the plan.
here, the flow is awkward - we create the CR, we create the PVC that should point to this CR, and we patch the CR to set the PVC as its owner. this suggests that the PVC is not really its owner - and think about it, let's say that I migrate a VM and the migration completed successfully, then I remove the VM and its PVCs - does it necessarily mean that I want to remove all the resources that are associated with the migration that has not been removed yet? I don't know, maybe not - I would still expect to, e.g., 'Get logs' to return the logs, including the CR of the migration, even though the VM was removed

Reply:
> > to enter the CleanUp, you need to either archive or cancel the migration. many users may just delete it without these steps. in that case - if the migration passed and the plan deleted, we will have leftovers, but it still somehow relate to the PVC-VM.
> 
> while we don't have empirical data on the amount of users remove plans without archiving them, this should have been addressed already by the message that appears when deleting a plan without archiving it first - there's a bold statement that this would result in leaving behind leftovers.

BTW, if that's really something you'd like to address - why not to add a validation webhook that blocks removing a plan that has not been archived?

Reply:
> > the ownership is just another way of cleanup.
> 
> does it imply that we would still initiate calls to remove the CRs when the migration is removed?
> 
no, there are two ways with this PR - calling CleanUp(using cancel/archive) or removing the PVC(or VM+disks).
of course we can call it by ourselves but then we will miss the logs (and conflicting the below discussion).
> > to enter the CleanUp, you need to either archive or cancel the migration. many users may just delete it without these steps. in that case - if the migration passed and the plan deleted, we will have leftovers, but it still somehow relate to the PVC-VM.
> 
> while we don't have empirical data on the amount of users remove plans without archiving them, this should have been addressed already by the message that appears when deleting a plan without archiving it first - there's a bold statement that this would result in leaving behind leftovers.
> 
> > in case they are deleted we are really not interested/need these so the ownership will take care of it.
> 
> yes, I see why you went this way - my issue with that is that the pvc is not really the owner of the CR. usually when you have true owner, you could specify it when creating a resource. for instance, when creating a migration you could say that the plan is its owner - the plan already exists, and the migration is derived by the plan. here, the flow is awkward - we create the CR, we create the PVC that should point to this CR, and we patch the CR to set the PVC as its owner. this suggests that the PVC is not really its owner - and think about it, let's say that I migrate a VM and the migration completed successfully, then I remove the VM and its PVCs - does it necessarily mean that I want to remove all the resources that are associated with the migration that has not been removed yet? I don't know, maybe not - I would still expect to, e.g., 'Get logs' to return the logs, including the CR of the migration, even though the VM was removed

This is correct. If we look on that like this way - yes, it may make sense to leave it as is. Letting the user get logs while the VM deleted.

Reply:
> > > to enter the CleanUp, you need to either archive or cancel the migration. many users may just delete it without these steps. in that case - if the migration passed and the plan deleted, we will have leftovers, but it still somehow relate to the PVC-VM.
> > 
> > 
> > while we don't have empirical data on the amount of users remove plans without archiving them, this should have been addressed already by the message that appears when deleting a plan without archiving it first - there's a bold statement that this would result in leaving behind leftovers.
> 
> BTW, if that's really something you'd like to address - why not to add a validation webhook that blocks removing a plan that has not been archived?

I think it's too "hard" to limit the deletion of the plan by archive it. We can - on deletion trigger the archive as a step of deletion. or, give a proper message about leftovers.

Reply:
> I think it's too "hard" to limit the deletion of the plan by archive it. We can - on deletion trigger the archive as a step of deletion. or, give a proper message about leftovers.

you mean it's too "strict", right? it shouldn't be hard..
I wouldn't spend the time on triggering 'archive' before removing the plan - it could have been nice to implement it with finalizers instead of introducing the 'archive' operation, but past contributors faced issues with this approach and ended up adding the 'archive' operation. I just don't see much value in doing 'archive' before 'delete' compared to block delete of plans that are not archived

> > > the ownership is just another way of cleanup.
> > 
> > 
> > does it imply that we would still initiate calls to remove the CRs when the migration is removed?
> 
> no, there are two ways with this PR - calling CleanUp(using cancel/archive) or removing the PVC(or VM+disks). of course we can call it by ourselves but then we will miss the logs (and conflicting the below discussion).
> 
> > > to enter the CleanUp, you need to either archive or cancel the migration. many users may just delete it without these steps. in that case - if the migration passed and the plan deleted, we will have leftovers, but it still somehow relate to the PVC-VM.

I lost you here but it seems irrelevant if we agree that setting the PVC as the owner of the CR is not the right way to go - so we'll need to remove the CRs as well when cleaning after the migration

Reply:
> > or, give a proper message about leftovers.

@liranr23 we already present a message about leftovers when deleting a plan without archiving it, why isn't it a proper message?

Reply:
> > > or, give a proper message about leftovers.
> 
> @liranr23 we already present a message about leftovers when deleting a plan without archiving it, why isn't it a proper message?

i didn't notice we do :)
it is proper as long we have one

Reply:
> while we don't have empirical data on the amount of users remove plans without archiving them, this should have been addressed already by the message that appears when deleting a plan without archiving it first - there's a bold statement that this would result in leaving behind leftovers.

ah ok, that's the message I referred to here ^